src/System.Management.Automation/engine/Subsystem/Commands/GetSubsystemCommand.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace System.Management.Automation.Subsystem
```

Comment:
If intention is to implement subsystems only internally (not by customers) could we make the new APIs `internal`? If no maybe the namespace should be `System.Management.Automation.Internal.Subsystem`?

Reply:
Some subsystems will be facing developers outside PowerShell teams, for example, `IPredictor`, which is the only subsystem kind defined today. We are partnering with another team in MS who is working on a Azure PowerShell command predictor.

The the suggestion framework (ref: #9115) could be another public facing extension point. We may make suggestion framework a subsystem, PowerShell can provide a default implementation, but there could also be custom implementations out there for specific scenarios.

Reply:
From previous comments I get understanding that intention is to make subsystems as a part of PowerShell Engine (no expectations are that customers will implement subsystems) and they are internal but every subsystem has a public interface for custom extensions.

Reply:
This is my previous comment on this. You might miss the "_maybe all except for ..._" part.

> To be honest, we are not expecting anyone outside the PowerShell team to implement most of the subsystems (maybe all except for `Prediction` and `Completion`).

Reply:
Thanks for clarify!

My thoughts are that perhaps it is better to implement (internally) a generic subsystem expandable with an public interface API.
We discuss something like in #10722.
It seems @PaulHigin was saying something like that about remoting https://github.com/PowerShell/PowerShell/issues/13282#issuecomment-665799880

Reply:
Paul's comment was exactly what we discussed in the committee meeting on how that can leverage the subsystem plugin model.